execution_workflows/TAPAS/modules/samtools.nf

```diff
@@ -0,0 +1,24 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process SAMTOOLS {
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process)) }
+
+    container "quay.io/biocontainers/samtools:1.13--h8c37831_0"
+
+    input:
+    tuple val(sample), path(bam), val(read_length)
+
+    output:
+    tuple val(sample), path("*.txt"), val(read_length), emit: read_coverage
+
+    script:
+    """
+    samtools depth $bam > read_coverage.txt    
+    """
+}
```

Comment:
Looks like this doesn't create a sample-specific coverage file, which is what we'd want when running tapas sample-by-sample. By that I mean I think we need something like:
```
script:
coverage_file="$sample" + "_read_coverage.txt"
"""
samtools depth $bam > $coverage_file
"""
```

I only see one output file in the results directory, which suggests the read_coverage.txt is overwritten when the pipeline is run (and I have no idea to which sample it corresponds).

![image](https://user-images.githubusercontent.com/49978382/190098775-13d3feb7-6dc0-4a60-beed-87ab7794176d.png)

Reply:
I have changed this